git_machete/client.py

```diff
@@ -1906,8 +1906,9 @@ def __sync_before_creating_pr(self) -> None:
         if current_branch not in self.managed_branches:
             self.add(current_branch)
 
-        if self.__is_merged_to_upstream(current_branch):
-            return
+        if self.__git.is_head_in_base_branch(current_branch, self.up_branch.get(current_branch)):
```

Comment:
Pls note that this check can be implemented with just `is_ancestor_or_equal`... no new code in git_operations.py needed 😅

Reply:
Lmk if a further clarification is needed on why _all commits from X are found in Y_ is equivalent to _X is ancestor of or equal to Y_